apollo-runtime/src/main/java/com/apollographql/apollo/ApolloCall.java

```diff
@@ -25,10 +27,22 @@
 
   @Nonnull ApolloCall<T> clone();
 
-  interface Callback<T> {
-    void onResponse(@Nonnull Response<T> response);
+  abstract class Callback<T> {
+    public abstract void onResponse(@Nonnull Response<T> response);
+
+    public abstract void onFailure(@Nonnull ApolloException e);
+
+    public void onHttpError(@Nonnull ApolloHttpException e) {
```

Comment:
Not sure about this, as it becomes too clumsy to have 3 params to handle one error.
And if user really decides to handle network error then don't see any issue to dig into `ApolloNetworkException` to extract what he needs.